### DISCUSSION TITLE
What is a softprompt? And how do I make one?

### DISCUSSION THREAD

**Discussion by pbosh** [oobabooga/text-generation-webui#230] [Q&A] [answered]
I've been using my own models for a few years now (hosted on my a100 racks in a colo) and I created a thing called a protected prompt. This essentially remains persistent and the chat uses the remaining tokens as available. I'm trying to recreate that with oogabooga. I think that's the idea behind a softprompt. Either way if anyone can point me in the direction of how to do this I'd be very grateful.

For clarity:
Protected prompt: This AI is a party assassin.
Rolling tokens: Hello/nHello/nWhat is the best time of year to boogie?

The protected prompt will always be present at the beginning of the tokens submitted. Rolling tokens drop off as the full amount of tokens fill up.

**Comment by oobabooga** [comment, marked as answer]
You can do it with an extension that defines an `input_modifier` like this:

```python
def input_modifier(string):
    """
    This function is applied to your text inputs before
    they are fed into the model.
    """ 

    return "This AI is a party assassin.\n\n" + string
```

https://github.com/oobabooga/text-generation-webui/wiki/Extensions#how-to-write-an-extension
